scanner/fetcher.go

```diff
@@ -171,7 +171,17 @@ func (f *Fetcher) genRanges(ctx context.Context) <-chan fetchRange {
 		defer close(ranges)
 		start, end := f.opts.StartIndex, f.opts.EndIndex
 
-		for start < end {
+		for start < end || f.opts.Continuous {
+			// In continuous mode wait for bigger STH every time we reach the end,
+			// including, possibly, the very first iteration.
+			if start == end { // Implies f.opts.Continuous == true.
+				if err := f.updateSTH(ctx); err != nil {
+					glog.Warningf("STH update cancelled: %v", err)
```

Comment:
Message could be clearer if there are other errors that can happen than the context being cancelled. Could mention it was waiting for log to increase in size.